src/placement/jump_map.c

```diff
@@ -705,8 +792,16 @@ get_object_layout(struct pl_jump_map *jmap, struct pl_obj_layout *layout,
 				D_DEBUG(DB_PL, "Target unavailable " DF_TARGET
 					". Adding to remap_list: fail cnt %d\n",
 					DP_TARGET(target), fail_tgt_cnt);
-				rc = remap_alloc_one(remap_list, k, target,
-						     false);
+
+				if (remap_grp_used == NULL) {
+					remap_grp_used = remap_gpu_alloc_one(&dgu_remap_list,
+									     dom_cur_grp_used);
+					if (remap_grp_used == NULL)
+						D_GOTO(out, rc = -DER_NOMEM);
+					realloc_grp_used = true;
+				}
```

Comment:
It's really hard to follow this, why don't you just simply allocate per-RDG "used domain bitmap" for all groups at the very beginning (Just like what we did for other bitmaps)?

Reply:
Hmm, I was trying to only allocate once for each layout generation normal, i.e. only re-allocate the grp tracking bitmap for remap case. Since this is on the hot path, I have to be cautious to avoid extra allocation for multiple group case.

Reply:
yes, I see your point now. Growing the 'uesd domain bitmap' on-demand could be better.